### DISCUSSION TITLE
Java+Selenide: Failure trying to access Grid with basic auth

### DISCUSSION THREAD

**Discussion by artashova** [selenide/selenide#2751] [Q&A]
Hello,
We are trying to secure Selenium standalone using Basic Auth.
In most cases, the RemoteWebDriver is accessible by using https://user:password@remote.url.example:4444
But Java requirements are different:
https://www.selenium.dev/documentation/grid/configuration/toml_options/#basic-auth-enabled

In case we use Selenide and configure remote URL by using
`Configuration.remote =  "https://user:password@remote.url.example:4444/wd/hub";`
it seems that Selenide ignores Java special requirements and browser initialisation fails (it fails the same way as if we call simple Selenium `WebDriver driver = new RemoteWebDriver(new URL("https://user:password@remote.url.example:4444"))`)

Is there any other way to configure Basic Auth for remote host in Selenide? 
Thank you in advance!

**Comment by asolntsev** [comment]
@artashova Selenide has its own tests that run Grid with BasicAuth. And adding credentials to the URL works fine there. See https://github.com/selenide/selenide/pull/2659/commits/e303e7b5f76814fbc08705a32e38dd4b534e64ee
